modules/standard/DateTime.chpl

```diff
@@ -433,8 +433,11 @@ module DateTime {
     timeStruct.tm_yday = (this - new date(year, 1, 1)).days: int(32);
 
     strftime(c_ptrTo(buf), bufLen, fmt.c_str(), timeStruct);
-    var str = createStringWithNewBuffer( __primitive("cast",
-                                                     c_string, c_ptrTo(buf)));
+    var str: string;
+    try! {
+      str = createStringWithNewBuffer( __primitive("cast",
```

Comment:
I know your PR is not about it but AFAIK this cast primitive can be replaced with just `c_ptrTo(buf):c_string`

Reply:
I can make that change here.